### DISCUSSION TITLE
Accept {crypto}selfhostedwebexplorer from config *file*

### DISCUSSION THREAD

**Discussion by Kixunil** [btcpayserver/btcpayserver#2768] [Ideas / Feature Requests]
## Problem

People who installed custom, self-hosted web-based explorer(s) currently need to remember to set override for them.

## Proposed solution

It'd be great if config file could contain a list of internal trusted explorers (or at least one) that'd would automatically set override unless user chose differently. Vendors could then make sure this setting is set whenever a self-hosted explorer is installed.

#### User installs BTCPayServer and doesn't touch overrides, some time later installs custom explorer

The config is updated by the vendor-provided tool and BTCPayServer picks it to override the setting

#### User installs BTCPayServer, changes override, later installs custom explorer

The config is updated by the vendor-provided tool but BTCPayServer does not pick it blindly.
The next time user opens BTCPayServer a message is shown that a custom explorer was installed and asks user whether to keep current setting or use the newly installed explorer.

#### User installs BTCPayServer and multiple explorers

The config is updated by the vendor-provided tool, BTCPayServer picks first by default. Overrides setting contains them in drop-down list, so it's easy to switch between them

### Cherry on top

To make UX even better the explorer links at invoices could have yellow warning sign (:warning:) icon if non-self-hosted explorer is used with hover text explaining privacy risk, if installed there would be a green checkmark (:heavy_check_mark:) and hover saying that it's safe because of self-hosted explorer.

## Context

I'd [love to provide this kind of automatic integration](https://github.com/debian-cryptoanarchy/cryptoanarchy-deb-repo-builder/issues/161) and I would prefer **not** to mess with database directly. Using config file would be safer.

**Comment by pavlenex** [comment]
AlI I want to to say is that I really appreciate the time you put into proposals and their formatting, makes it so easy to understand the idea when people take time to communicate it properly.

**Reply by Kixunil** [reply]
Glad to see it's as helpful as I hoped. I do it exactly because I know the difference myself. :) And thanks for the feedback, so that I know my effort is not wasted and I should continue.

**Reply by ghost** [reply]
That's why Free Software is winning. :black_heart:
